test/unit/mod/diam_test.mod

```diff
@@ -0,0 +1,27 @@
+NEURON {
+    SUFFIX diam_test
+}
+
+PARAMETER {
+    diam
+}
```

Comment:
Should we permit `diam` as a `PARAMETER`? I see that `diam` is removed from the parameter list in `modcc/module.cpp`.

What does NMODL do if a user creates an ASSIGNED or PARAMETER called `diam` and sets  its value explicitly? Will that override the real value? If NEURON doesn't need these magic variables to be explicitly declared, then we could enforce that they are not without forcing users to write NMODL that is incompatible with NEURON.